File: #104.Maximum Depth of Binary Tree/recursion/Maximum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
//递归解法
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(!root)
            return 0;
        else{
            int left_length=maxDepth(root->left);
            int right_length=maxDepth(root->right);
            return left_length>right_length?left_length+1:right_length+1;
        }
    }
};

File: algorithms/104.Maximum Depth of Binary Tree/Maximum Depth of Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function (root) {

    return length(root);

    function length(node) {
        if (!node) {
            return 0;
        }
        return Math.max(length(node.left), length(node.right)) + 1;
    }
};